url7b90c1010b_DastanIqbal_HelloZMQwithJava_tgz-pJ8-ClientJ8/src/zmq/Mtrie.java

```diff
@@ -0,0 +1,394 @@
+    {
+        if (prefix_ == null || prefix_.length == start_) {
+            if (pipes != null) {
+                boolean erased = pipes.remove(pipe_);
+                assert (erased);
+                if (pipes.isEmpty ()) {
+                    pipes = null;
+                }
+            }
+            return pipes == null;
+        }
+
+        byte c = prefix_[ start_ ];
+        if (count == 0 || c < min || c >= min + count)
+            return false;
+
+        Mtrie next_node =
+            count == 1 ? next[0] : next[c - min];
+
+        if (next_node == null)
+            return false;
+
+        boolean ret = next_node.rm_helper (prefix_ , start_ + 1, pipe_);
+        if (next_node.is_redundant ()) {
+            assert (count > 0);
+
+            if (count == 1) {
+                next = null;
+                count = 0;
+                --live_nodes;
+                assert (live_nodes == 0);
+            }
+            else {
+                next[c - min] = null ; 
+                assert (live_nodes > 1);
+                --live_nodes;
+
+                //  Compact the table if possible
+                if (live_nodes == 1) {
+                    //  If there's only one live node in the table we can
+                    //  switch to using the more compact single-node
+                    //  representation
+                    int i;
+                    for (i = 0; i < count; ++i) {
+                        if (next[i] != null) {
+                            break;
+                        }
+                    }
+
+                    assert (i < count);
+                    min += i;
+                    count = 1;
+                    Mtrie old = next [i];
+                    next = new Mtrie [] { old };
+                }
+                else if (c == min) {
+                    //  We can compact the table "from the left"
+                    int i;
+                    for (i = 1; i < count; ++i) {
+                        if (next[i] != null) {
+                            break;
+                        }
+                    }
+                    
+                    assert (i < count);
+                    min += i;
+                    count -= i;
+                    next = realloc (next, count, true);
+                }
+                else if (c == min + count - 1) {
+                    //  We can compact the table "from the right"
+                    int i;
+                    for (i = 1; i < count; ++i) {
+                        if (next[count - 1 - i] != null) {
+                            break;
+                        }
+                    }
+                    assert (i < count);
+                    count -= i;
+                    next = realloc(next, count, false);
+                }
+            }
+        }
+
+        return ret;
+    }
+
+    //  Signal all the matching pipes.
+    public void match(byte[] data_, int size_, IMtrieHandler func_, Object arg_) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.